clients/src/main/java/org/apache/kafka/common/Cluster.java

```diff
@@ -28,13 +28,14 @@
  */
 public final class Cluster {
 
+    private final boolean isBootstrapConfigured;
     private final List<Node> nodes;
     private final Set<String> unauthorizedTopics;
     private final Map<TopicPartition, PartitionInfo> partitionsByTopicPartition;
     private final Map<String, List<PartitionInfo>> partitionsByTopic;
     private final Map<String, List<PartitionInfo>> availablePartitionsByTopic;
-    private final Map<Integer, List<PartitionInfo>> partitionsByNode;
-    private final Map<Integer, Node> nodesById;
+    private final Map<String, List<PartitionInfo>> partitionsByNodeId;
+    private final Map<String, Node> nodesById;
```

Comment:
I needed to have `nodesById` use a String since I needed to be able to get the `Node` from a RequestSend object, which only has access to the string version. I changed the other one as well since it looked a little funny being the only one using an integer key. One nice refactor might be to wrap  merge these into a single map with a container object containing the different values.